Bottom: dff5aa9b300359119c273862f03db311aa8cd6f2
Top:    b75195dc98a7597414a8a3724767a655e7f89be7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-10 03:47:40 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index bb52616..d404204 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -308,6 +308,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
